### PR TITLE
overlord/servicestate: restart the relevant journald service when a journal quota group is modified

### PR DESCRIPTION
**Pull request by Meulengracht**
This was missing and discovered while adding this case to a spread test. It's not that critical, but it still needs to be done. This can land after https://github.com/snapcore/snapd/pull/11815 and includes that PR.

### CODE REVIEW

**Review by pedronis — APPROVED** [review]
last commit looks alright to me

**Comment by pedronis** [comment]
@Meulengracht btw is this going to be tested by the spread test?

**Comment by Meulengracht** [comment]
> @Meulengracht btw is this going to be tested by the spread test?

Yes! Exactly this scenario is tested in PR https://github.com/snapcore/snapd/pull/11916, this was how I discovered the issue

File: overlord/servicestate/quota_handlers.go [hunk 1]
```diff
@@ -592,6 +594,15 @@ func ensureSnapServicesForGroup(st *state.State, t *state.Task, grp *quota.Group
 					}
 				}
 			}
+
+			// If the journal has changed (i.e old and new not being empty) then we
+			// need to restart the journal for that namespace. For other cases
+			// this is not necessary, as either the journal daemon will be started or
+			// stopped as a part of removal.
+			if old != "" && new != "" {
```
[the remark this answers is not in the document]
Comment (mvo5) [reply overlord/servicestate/quota_handlers.go:602]: I guess it is impossible but wanted to double check :)

**Review by mvo5 — APPROVED** [review]

**Pull request merged**